### PR TITLE
Sorting an array of 0's and 1's

### PR DESCRIPTION
**Pull request by iaditichine**
I have done all the required changes in the code and updated readme.md file for issue- #3901.

### CODE REVIEW

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/* Sorting an array of 0's and 1's
+
+Given an array of size n which contains only 0's and 1's
+in any order,we have to modify the array in such a way
+that 0's are on left side and 1's on the right side.
+Test case-1:
+Input-1:
+N = 5
+arr[] = {0, 0, 1, 1, 0}
+Output-1: 0 0 0 1 1
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Remove redundant comments

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+*/
+
+#include <iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Replace header file with `#include <bits/stdc++.h>`

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    	//printing 1 for c1 times
+        cout << 1 << " "; 
+    }
+    cout << endl;
+}
+int main()
+{
+    int T;
+    cin >> T;
+    while (T--)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: As we don't have online judge so remove iterating test cases.

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+Output: 0 0 0 0
+There are no 1's in the given array,
+so the modified array is 0 0 0 0.
+
+Test case-2:
+Input:
+N==6
+arr[]={0,1,1,0,1,0}
+Output:0 0 0 1 1 1
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Remove this type of I/P and O/P, Also use proper display message for I/P and O/P

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Use proper naming convention for variable names don't use single letters
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp (hunk 1 above)
Comment (iaditichine) [reply C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp:10]: I have added sample input output at the end of code. Should I change it?

**Review by ankitaggarwal23 — COMMENTED** [review]
As I found that you're spamming by opening PR for same issue. Kindly don't repeat this otherwise we have to take action in this.

**Comment by iaditichine** [comment]
I have no intention of Spamming. I create a new PR everytime because you request some changes and I don't know any other way of doing changes in same PR.

**Comment by iaditichine** [comment]
I have no intention of spamming. It's just because I'm new to open source. Sorry.

**Comment by ankitaggarwal23** [comment]
> I have no intention of spamming. It's just because I'm new to open source. Sorry.

If you're facing any issue just let us know but don't do it. Great

**Comment by iaditichine** [comment]
@ankitaggarwal23 , can you tell me how to make changes in the same PR?

**Comment by ankitaggarwal23** [comment]
> @ankitaggarwal23 , can you tell me how to make changes in the same PR?

Go the branch which you have created for this PR and do the changes and push you changes?

**Comment by ankitaggarwal23** [comment]
> > @ankitaggarwal23 , can you tell me how to make changes in the same PR?
> 
> Go the branch which you have created for this PR and do the changes and push you changes?

If you still stuck DM me on discord I will help you!

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    for (int i = 0; i < size_of_array; i++)
+    {
+        cin >> array[i];
+    }
+    /*  Two counters to count number of 0's and 1's respectively
+	    which are initialized to 0*/
+    int counter_for_0 = 0, counter_for_1 = 0;
+    for (int i = 0; i < size_of_array; i++)
+    {
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: remove the space

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+that 0's are on left side and 1's on the right side.
+
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+//Function to sort an array of 0's and 1's
+void sort()
+{
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: remove the space

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        cout << 0 << " "; 
+    }
+    while (counter_for_1--)
+    {
+    	
+        cout << 1 << " "; 
+    }
+    cout << endl;
+}
+int main()
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp:47]: Add a new line above main

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+array[] = {0, 0, 0, 0}
+Output: 0 0 0 0
+There are no 1's in the given array,
+so the modified array is 0 0 0 0.
+
+
+Input:
+size_of_array==6
+array[]={0,1,1,0,1,0}
+Output:0 0 0 1 1 1
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Remove extra test case one is enough also use proper display message for I/p and o/p

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp (hunk 8 above)
Comment (iaditichine) [reply C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: What kind of message?

Comment (ankitaggarwal23) [reply C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Use cout for proper displaying message and write in the same way your code is producing outputs.

Comment (iaditichine) [reply C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Okay, will do it.

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        cin >> array[i];
+    }
+    /*  Two counters to count number of 0's and 1's respectively
+	    which are initialized to 0*/
+    int counter_for_0 = 0, counter_for_1 = 0;
+    for (int i = 0; i < size_of_array; i++)
+    {
+        if (array[i] == 0)
+        {
+		    counter_for_0++;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: Remove the one tab space and align it with continue;

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp (hunk 9 above)
Comment (ankitaggarwal23) [reply C-Plus-Plus/other/Sorting_an_array_of_zeroes_and_ones.cpp]: rest looks good to merge

**Comment by iaditichine** [comment]
@ankitaggarwal23 , I have made the change.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**